### ISSUE TITLE
Ditch django_compressor in favour of a Node-based bundling setup

### ISSUE THREAD

**Issue by gasman** [torchbox/wagtail#2124]
As of version 2.0, django_compressor now depends on the packages `rjsmin` and `rcssmin`, which include native extensions and are thus non-trivial to install on Windows (see https://groups.google.com/d/msg/wagtail/-CDQcHk5F9M/aL_DbhjdCQAJ). Additionally, there have been reports of bugs in rjsmin's compression (specifically, SimpleMDE uses the regex `/^ *-+:*$/` which rjsmin incorrectly compresses to `/^*-+:*$/`), and our current django_compressor setup doesn't allow specific addons/apps using the `insert_editor_js` hook to opt out of compression. Finally, it's an extra thing that end users of Wagtail need to configure (or at least, be aware of) when deploying.

The main reason for using django_compressor (SASS compilation) has already been superseded by our Node toolchain, and minification/bundling is something else that could certainly be done just as well (or better) through Node. The one thing we'd lose would be the dynamic 'discovery' of bundled assets based on `{% compress %}` tags - and thus the ability for third-party modules to insert themselves into those bundles via `insert_editor_js` - but it's probably a good thing for maintainability for our bundles to be defined statically in a gulp config (or similar), and for third-party modules to deal with minification themselves if they consider it important.

**Comment by davecranwell** [comment]
See also #1143

**Comment by spapas** [comment]
This may be a little out of date, however, for people that would still like to use django-compressor in Windows, I'd like to point out that you can easily install it using the ``--install-option="--without-c-extensions"`` pip parameter:

```
(venv) C:\progr\py\test>pip install --install-option="--without-c-extensions" django-compressor
You are using pip version 6.1.1, however version 8.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting django-compressor
  Using cached django_compressor-2.0-py2.py3-none-any.whl
Collecting django-appconf>=0.4 (from django-compressor)
  Using cached django_appconf-1.0.1-py2.py3-none-any.whl
Collecting rcssmin==1.0.6 (from django-compressor)
  Using cached rcssmin-1.0.6.tar.gz
Collecting rjsmin==1.0.12 (from django-compressor)
  Using cached rjsmin-1.0.12.tar.gz
Collecting six (from django-appconf>=0.4->django-compressor)
  Using cached six-1.10.0-py2.py3-none-any.whl
Installing collected packages: six, django-appconf, rcssmin, rjsmin, django-compressor
  Running setup.py install for rcssmin
  Running setup.py install for rjsmin
Successfully installed django-appconf-1.0.1 django-compressor-2.0 rcssmin-1.0.6 rjsmin-1.0.12 six-1.10.0
```

This is just for completeness, I also think that moving to node-js tools for building your js/css is the way to go :+1:

**Comment by thearrow** [comment]
This is awesome. Thanks so much for your great work on this! :+1:  Any idea when this will make it into a release? (I was having some issues with django-compressor in my docker-based deploy setup, so right now I'm pulling and building `master` from github :open_mouth: )

**Comment by gasman** [comment]
Fixed in #2132
